### PR TITLE
Closes #7: Store isAdmin flag on user entity and assign role upon accept

### LINKED ISSUE #7: Store isAdmin flag on user entity and assign role upon accept
Being an organization admin is currently derived from the user's actual roles.  This should be persisted as a flag on the user entity.  Assignment of this admin role should only happen when a user accepts the organization invite

Summary of required changes:
* Store as flag on entity
* expose this flag instead of derived role based flag
* don't assign admin role immediately but only after accepting (or directly in case the user is in status `ACCEPTED` already)

This will also fix 400 returns on assigning admin role to an unexisting user (i.e. not yet signed up).

### PR DESCRIPTION
**Pull request by steinmaerivoet**

### CODE REVIEW

File: src/main/java/com/ixortalk/organization/api/rest/UserRestController.java [hunk 1]
```diff
@@ -93,6 +97,10 @@
                     userRestResource.save(user.accepted());
                     organizationCallbackAPI.userAccepted(new UserInOrganizationDTO(user.getLogin(), organizationRestResource.findByUsers(user).map(Organization::getId).orElseThrow(BadRequestException::new)));
                     auth0Users.unblockUser(user.getLogin());
+                    if (user.isAdmin()) {
+                        Organization organization = organizationRestResource.findByUsers(user).orElseThrow(() -> new IllegalStateException("Error while accepting invite. User is not added to inviting organization."));
+                        userService.assignAdminRole(organization, user);
```
Comment (wjans) [inline src/main/java/com/ixortalk/organization/api/rest/UserRestController.java:102]: This is not tested.

**Review by wjans — CHANGES_REQUESTED** [review]
Make sure all changes are tested.

**Review by wjans — APPROVED** [review]

File: src/main/java/com/ixortalk/organization/api/rest/OrganizationRestController.java [hunk 2]
```diff
@@ -131,7 +131,7 @@
                                 organization
                                         .getUsers()
                                         .stream()
-                                        .filter(user -> enhanceUserService.isAdmin(user))
+                                        .filter(User::isAdmin)
```
Comment (jraeymaek) [inline src/main/java/com/ixortalk/organization/api/rest/OrganizationRestController.java:134]: @steinmaerivoet @wjans I think this change is breaking, unless the isAdmin column is kept in sync with auth0 before the release.
Are you aware of that?

Comment (wjans) [reply src/main/java/com/ixortalk/organization/api/rest/OrganizationRestController.java:134]: Yes, this is breaking indeed.  This PR has been merged and had a minor version bump upon realeasing (version `1.1.0`).  Minor version bumps in the IxorTalk platform indicate that changes are not backwards compatible.

Comment (wjans) [reply src/main/java/com/ixortalk/organization/api/rest/OrganizationRestController.java:134]: Another thing that has changed/improved here is that users will also be exposed as being an admin, even before they accepted their invitation.  However, both `status` and `isAdmin` are exposed, so it's up to clients to deal with this properly.

**Pull request merged**